client/src/www/app/outline_server_repository/vpn.ts

```diff
@@ -0,0 +1,45 @@
+// Copyright 2024 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+export interface ShadowsocksSessionConfig {
+  host?: string;
+  port?: number;
+  password?: string;
+  method?: string;
+  prefix?: string;
+}
+
+export const enum TunnelStatus {
+  CONNECTED,
+  DISCONNECTED,
+  RECONNECTING
+}
+
+// VpnApi is how we talk to the platform-specific VPN API.
+export interface VpnApi {
```

Comment:
Do we prefer `VPNService`? `VpnApi` sounds very generic.

Reply:
I thought that "service" gets confusing with each individual vpn service. So API was more broad and better aligned with the concept it represents.

Reply:
I've played with this a bit, and I prefer VpnApi for a few reasons:

- A "VPN" is one network. In our case one service. This is not representing a VPN, it's used for all VPNs.
- This is not a domain model/product concept. It's an application/infrastructure detail, and it encapsulates the VPN api of the platform. So VpnApi seems aligned.